cfspopcon/algorithm_class.py

```diff
@@ -128,19 +137,63 @@ def wrapped_function(**kwargs: Any) -> dict:
             for i, key in enumerate(return_keys):
                 if skip_unit_conversion:
                     result_dict[key] = result[i]
+                elif isinstance(result[i], float):
+                    result_dict[key] = convert_to_default_units(Quantity(result[i], ureg.dimensionless), key)
                 else:
                     result_dict[key] = convert_to_default_units(result[i], key)
 
             return result_dict
 
         return cls(wrapped_function, return_keys, name=name)
 
+    @classmethod
+    def register_algorithm(
+        cls, return_keys: list[str], name: Optional[str] = None, skip_unit_conversion: bool = False
+    ) -> GenericFunctionType:
+        """Decorate a function and turn it into an Algorithm. Usage: @Algorithm.register_algorithm(return_keys=["..."])."""  # noqa: D402
+
+        def function_wrapper(func: GenericFunctionType) -> GenericFunctionType:
+            Algorithm.from_single_function(func, return_keys=return_keys, name=name, skip_unit_conversion=skip_unit_conversion)
```

Comment:
Should we check if `name is None` and then use `func.__name__` instead?

Reply:
Resolve in #41